### PR TITLE
Local plot schema

### PR DESCRIPTION
**Pull request by theengineear**
This PR stops auto-loading the plot-schema on import. We decided in #642 that this shouldn't need to force a major version bump, so I'll add a minor version bump here.

See #614 for the original PR. I noticed that we actually had a weird PY3 conflict due to a change in how importing works in PY3 (See commit notes).

### CODE REVIEW

**Comment by theengineear** [comment]
@Kully can you give this a review? We discussed the no-download-on-import portion of this in #642, but I've also done the following (which we *haven't* discussed wrt Semantic Versioning):

* `.js` and `.json` files --> `plotly/package_data`. I can't imagine any users are using the non-python files directly in scripts.
* auto-object generation. This isn't backwards-incompatible. In fact, it's exactly what we had before, except now it's not generated on import in memory.

@chriddyp, 👍 on the meta programming introduced in 51ef008?

File: plotly/graph_objs/graph_objs_tools.py [hunk 1]
```diff
@@ -37,7 +37,8 @@ def _list_help(object_name, path=(), parent_object_names=()):
     items = graph_reference.ARRAYS[object_name]['items']
     items_classes = [graph_reference.string_to_class_name(item)
                      for item in items]
-    lines = textwrap.wrap(repr(items_classes), width=LINE_SIZE-TAB_SIZE)
+    items_classes.sort()
+    lines = textwrap.wrap(repr(items_classes), width=LINE_SIZE-TAB_SIZE - 1)
```
Comment (Kully) [inline plotly/graph_objs/graph_objs_tools.py]: Nit Picky, but would `lines = textwrap.wrap(repr(items_classes), width=LINE_SIZE - TAB_SIZE - 1)`
or `lines = textwrap.wrap(repr(items_classes), width=LINE_SIZE-TAB_SIZE-1)` be better for consistency?

File: plotly/graph_objs/graph_objs_tools.py [hunk 2]
```diff
@@ -167,7 +169,7 @@ def _dict_attribute_help(object_name, path, parent_object_names, attribute):
                 if isinstance(val, list) and attribute == 'showline':
                     val = val[0]
 
-                lines = textwrap.wrap(val, width=LINE_SIZE)
+                lines = textwrap.wrap(val, width=LINE_SIZE - 1)
```
Comment (Kully) [inline plotly/graph_objs/graph_objs_tools.py]: See [previous comment](https://github.com/plotly/plotly.py/pull/648/files#r94804612)

File: plotly/graph_objs/graph_objs_tools.py [hunk 3]
```diff
@@ -54,9 +55,10 @@ def _list_help(object_name, path=(), parent_object_names=()):
 
 def _dict_object_help(object_name, path, parent_object_names):
     """See get_help()."""
-    attributes = graph_reference.get_valid_attributes(object_name,
-                                                      parent_object_names)
-    lines = textwrap.wrap(repr(list(attributes)), width=LINE_SIZE-TAB_SIZE)
+    attributes = list(
+        graph_reference.get_valid_attributes(object_name, parent_object_names))
+    attributes.sort()
+    lines = textwrap.wrap(repr(list(attributes)), width=LINE_SIZE-TAB_SIZE - 1)
```
Comment (Kully) [inline plotly/graph_objs/graph_objs_tools.py]: See [previous comment](https://github.com/plotly/plotly.py/pull/648/files#r94804612)

File: update_graph_objs.py [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+FLAG = '# AUTO-GENERATED BELOW. DO NOT EDIT! See makefile.'
+
+
+def get_non_generated_file_lines():
+    """
+    Copy each line up to our special FLAG line and return.
+
+    :raises: (ValueError) If the FLAG isn't found.
+    :return: (list) The lines we're copying.
+
```
Comment (Kully) [inline update_graph_objs.py]: I think the doc strings should not have the extra space between last line and the triple quotes """

File: update_graph_objs.py [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+    with open('./plotly/graph_objs/graph_objs.py', 'r') as f:
+        for line_to_copy in f:
+            if line_to_copy.startswith(FLAG):
+                flag_found = True
+                break
+            lines_to_copy.append(line_to_copy)
+    if not flag_found:
+        raise ValueError(
+            'Failed to find flag:\n''{}\n''in graph_objs_tools.py.'
+            .format(FLAG)
```
Comment (Kully) [inline update_graph_objs.py]: More a suggestion, but for legibility, what do you think about putting the `format(FLAG)` on the line above? I think it's 80 chars on that line, but would be worth IMO

File: update_graph_objs.py [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+        )
+    return lines_to_copy
+
+
+def print_figure_patch(f):
+    """Print a patch to our Figure object into the given open file."""
```
Comment (Kully) [inline update_graph_objs.py]: For one-liner doc strings, which is more common to your knowledge:

```one-line-here```

```
one-line-here
```

I usually do the latter. Let me know if you think this is worth worrying/thinking about

File: update_graph_objs.py (hunk 4 above)
Comment (theengineear) [inline update_graph_objs.py]: Hehe, I disagree on this one. But I'm OK with changing it. You like to do the blank line *after* the `"""` then?

File: update_graph_objs.py (hunk 5 above)
Comment (theengineear) [inline update_graph_objs.py]: Huh... that string looks wrong to me, the extra `'` characters? I'll fix this up.

File: update_graph_objs.py (hunk 6 above)
Comment (theengineear) [inline update_graph_objs.py]: One-liners can/should be one-liners:

https://www.python.org/dev/peps/pep-0257/#one-line-docstrings

**Comment by theengineear** [comment]
@Kully alright with this?

**Comment by theengineear** [comment]
@chriddyp , it'd be great to get a high-level 👍 from you on 51ef008 still.

You don't need to dig into the code very much. In general I'm just wondering if you think the approach of moving the meta programming we were *already* doing into auto-generated lines inside the `graph_objs.py` files seems OK.

Specifically, all the work is done in `update_graph_objs.py`, where I literally print new lines into `graph_objs.py` programmatically.

**Comment by theengineear** [comment]
Since I've got a lot of moving parts happening in this repo, I'm going to merge after chatting with @Kully in slack and with @chriddyp's 👍 for hard-coding our graph_objs.

**Pull request merged**